stac_fastapi/core/stac_fastapi/core/core.py

```diff
@@ -507,13 +507,26 @@ async def all_catalogs(
                 break
             token = next_token
 
+        if catalog_path:
+            catalog_url = f"catalogs/{catalog_path}"
+        else:
+            catalog_url = ""
+
         links = [
-            {"rel": Relations.root.value, "type": MimeTypes.json, "href": base_url},
-            {"rel": Relations.parent.value, "type": MimeTypes.json, "href": base_url},
+            {
+                "rel": Relations.root.value,
+                "type": MimeTypes.json,
+                "href": urljoin(base_url, catalog_url),
+            },
+            {
+                "rel": Relations.parent.value,
+                "type": MimeTypes.json,
+                "href": urljoin(base_url, catalog_url),
```

Comment:
This doesn't look quite right, either - this should be the parent Catalog but contains what the `self` link should contain.

Reply:
As this is the `/catalogs` endpoint it returns a list of catalogs, with each one containing the expected self and parent links. This link section is only for the overall `/catalogs` response. So the self endpoint is in fact `/catalogs` and the parent link would then be to the catalogue itself. Is this the correct behaviour? For example see [here](https://[dev.eodatahub.org.uk/api/catalogue/stac/catalogs/supported-datasets/catalogs](https://dev.eodatahub.org.uk/api/catalogue/stac/catalogs/supported-datasets/catalogs))